### PR TITLE
Display a message if hardware rendering is supported

### LINKED ISSUE #13336: Default to software rendering on Android Emulators
Android supports software rendering but it is currently opt-in (via the `--enable-software-rendering` engine flag).

Due to numerous rendering issues in emulated environments, we should consider switching to software rendering on emulators on Android. We already do this on iOS simulators.

So far, we have had issues around reliably detecting (at runtime) if the device is an emulator.

However, the tools can check if the `ro.kernel.qemu` property is set to 1 and have the tools automatically add the `--enable-software-rendering` flag. For users who have correctly configured their emulators to use the host GPU, this will probably cause a slowdown in rendering. However, software rendering is way more reliable and leads to fewer surprises. We can still allow (via the tools, as the default is always GPU rendering) the user to explicitly use GPU rendering.

### PR DESCRIPTION
**Pull request by jonahwilliams**
If the device is an android emulator, display a message depending on whether software/hardware rendering is enabled.

Fixes #13336

### CODE REVIEW

**Review by yjbanov — APPROVED** [review]

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 1]
```diff
@@ -284,8 +284,26 @@ class RunCommand extends RunCommandBase {
...
 
     for (Device device in devices) {
-      if (await device.isLocalEmulator && !isEmulatorBuildMode(getBuildMode()))
-        throwToolExit('${toTitleCase(getModeName(getBuildMode()))} mode is not supported for emulators.');
+      if (await device.isLocalEmulator) {
+        if (await device.supportsHardwareRendering) {
+          final bool enableSoftwareRendering = argResults['enable-software-rendering'] == true;
+          if (enableSoftwareRendering) {
+            printStatus(
+              'Using software rendering with device ${device.name}. You may get better performance'
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/run.dart]: you're missing a space between "performance" and "with" in this string literal, so it'll render as "better performancewith hardware".

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 2]
```diff
@@ -284,8 +284,26 @@ class RunCommand extends RunCommandBase {
...
+              'with hardware mode by configuring hardware rendering for your device.'
+            );
+          } else {
+            printStatus(
+              'Using hardware rendering with device ${device.name}. If you get graphics artifacts,'
+              'consider enabling software rendering with "--enable-software-rendering".'
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/run.dart]: similar here

**Comment by Hixie** [comment]
Oh, I forgot to suggest that maybe a test would be in order, too. :-)

**Comment by jonahwilliams** [comment]
Shoot, I missed these.  Will do a follow up now

**Pull request merged**